pkg/apis/pipeline/v1alpha1/image_resource.go

```diff
@@ -76,11 +78,26 @@ func (s *ImageResource) Replacements() map[string]string {
 	}
 }
 
+// GetUploadContainerSpec returns the spec for the upload container
 func (s *ImageResource) GetUploadContainerSpec() ([]corev1.Container, error) {
 	return nil, nil
 }
+
+// GetDownloadContainerSpec returns the spec for the download container
 func (s *ImageResource) GetDownloadContainerSpec() ([]corev1.Container, error) {
 	return nil, nil
 }
+
+// SetDestinationDirectory sets the destination for the resource
 func (s *ImageResource) SetDestinationDirectory(path string) {
 }
+
+// GetOutputImageDir return the path to get the index.json file
+func (s *ImageResource) GetOutputImageDir() string {
+	return s.OutputImageDir
+}
+
+func (s ImageResource) String() string {
+	json, _ := json.Marshal(s)
```

Comment:
are you sure we want to ignore the error here? if so maybe put a comment explaining why - I would think we'd want to do something different if it failed tho (at least log maybe?) 🤔

Reply:
the String() func works as a toString func to return the contents as a string and has to follow the interface and therefore cannot return an error
if the `Marshal` func gives an error, the returned string will be empty

Reply:
okay! thanks for including the comment